.github/workflows/create-draft-release.yml

```diff
@@ -0,0 +1,63 @@
+name: "Create Draft Release"
+
+env:
+  BUILDDIR: "build"
+  DEVELOPER_DIR: "/Applications/Xcode_12.5.1.app/Contents/Developer"
+
+on:
+  workflow_dispatch
+
+concurrency: 
+  group: publish-release-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    name: "Publish binaries for release"
+    runs-on: macos-11
+
+    steps:
+      - name: "Checkout sources"
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - name: "Build release distribution"
+        run: make release
+        env:
+          GITHUB_ACTOR: ${{ github.actor }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Extract version information from project"
+        run: |
+          MARKETING_VERSION=$(xcrun xcodebuild -showBuildSettings -project Sparkle.xcodeproj | grep MARKETING_VERSION | cut -f 7 -d ' ')
+          echo ::set-env name=MARKETING_VERSION::${MARKETING_VERSION}
```

Comment:
I had just found this article on google this but is this going to be a problem setting env variables like this? https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

Reply:
One other approach is that we pass MARKETING_VERSION and PROJECT_CURRENT_VERSION as input in the GitHub workflow after all, and then the workflow overrides those values in CommonConfig.xcconfig and `git add`'s that change. I'm fine with either approach.

Reply:
Good catch. I've switched to the new [environment files](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#environment-files) approach. 

> is setting an env variable necessary or there is no other way to pass info from one step to another?

There are other methods, but this was the simplest way of getting simple key value pairs across steps in the workflow. 

I think the question I'd ask in response to the suggestions here is: do you want to allow unvalidated free text input to determine the contents of MARKETING_VERSION and PROJECT_CURRENT_VERSION? I guess it really doesn't make much difference - my main issue here is that I'm wary of committing changes from within CI workflows.